hi-ml-cpath/src/health_cpath/preprocessing/loading.py

```diff
@@ -71,49 +138,222 @@ class LoadROId(MapTransform):
     - `SlideKey.FOREGROUND_THRESHOLD` (float): threshold used to segment the foreground
     """
 
-    def __init__(self, reader: WSIReader, image_key: str = SlideKey.IMAGE, level: int = 0,
-                 margin: int = 0, foreground_threshold: Optional[float] = None) -> None:
+    def __init__(
+        self, image_key: str = SlideKey.IMAGE, foreground_threshold: Optional[float] = None, **kwargs: Any
+    ) -> None:
         """
-        :param reader: And instance of MONAI's `WSIReader`.
         :param image_key: Image key in the input and output dictionaries.
-        :param level: Magnification level to load from the raw multi-scale file.
-        :param margin: Amount in pixels by which to enlarge the estimated bounding box for cropping.
         :param foreground_threshold: Pixels with luminance below this value will be considered foreground.
         If `None` (default), an optimal threshold will be estimated automatically using Otsu's method.
+        :param kwargs: Additional arguments for `BaseLoadROId`.
         """
-        super().__init__([image_key], allow_missing_keys=False)
-        self.reader = reader
-        self.image_key = image_key
-        self.level = level
-        self.margin = margin
+        MapTransform.__init__(self, [image_key], allow_missing_keys=False)
+        BaseLoadROId.__init__(self, image_key=image_key, **kwargs)
         self.foreground_threshold = foreground_threshold
 
-    def _get_bounding_box(self, slide_obj: 'CuImage') -> Tuple[box_utils.Box, float]:
-        # Estimate bounding box at the lowest resolution (i.e. highest level)
-        highest_level = slide_obj.resolutions['level_count'] - 1
-        scale = slide_obj.resolutions['level_downsamples'][highest_level]
-        slide = load_slide_at_level(self.reader, slide_obj, level=highest_level)
+    def _load_slide_at_level(self, slide_obj: _OpenSlideOrCuImage, level: int) -> np.ndarray:
+        """Load full slide array at the given magnification level.
+
+        This is a manual workaround for a MONAI bug (https://github.com/Project-MONAI/MONAI/issues/3415)
+        fixed in a currently unreleased PR (https://github.com/Project-MONAI/MONAI/pull/3417).
```

Comment:
@dccastro can we safely remove this workaround? I see that you were the one who opened the issue